### PR TITLE
Making the net_spec python3 compatible

### PR DESCRIPTION
**Pull request by philkr**
Fixes a few python3 bugs caused by iteritem and input redirection.

### CODE REVIEW

File: examples/pycaffe/caffenet.py [hunk 1]
```diff
@@ -45,10 +46,10 @@ def caffenet(lmdb, batch_size=256, include_acc=False):
 
 def make_net():
     with open('train.prototxt', 'w') as f:
-        print >>f, caffenet('/path/to/caffe-train-lmdb')
+        print( caffenet('/path/to/caffe-train-lmdb'), file=f )
```
Comment (longjon) [inline examples/pycaffe/caffenet.py]: Please follow PEP8 here and below: http://legacy.python.org/dev/peps/pep-0008/#pet-peeves.

**Comment by longjon** [comment]
I guess this adds `six` as a dependency, which has been suggested before. That seems pretty lightweight and useful, so I have no problem with that, but this PR should update https://github.com/BVLC/caffe/blob/master/python/requirements.txt.

Other than that and the style issue noted, this looks good to go. Thanks for keeping us up with the times @philkr!

**Comment by philkr** [comment]
I updated the PR.

**Comment by longjon** [comment]
Looks good, thanks.

**Pull request merged**